donut-app/source/dunkinLocator.mc

```diff
@@ -0,0 +1,281 @@
+	}
+	
+	// Updates the position variable according to the info
+	function gotPositionCallback(info) {
+		System.println("Updating position to " + info.position.toDegrees());
+		posDegrees = info.position.toDegrees();
+    }
+    
+    // Signals that this locator should cease listening to location NOR should it periodically look for the nearest Dunkin
+    function stopListening() {
+		Position.enableLocationEvents(Position.LOCATION_DISABLE, null);
+		posDegrees = null;
+		
+		if(requestTimer != null) {
+			requestTimer.stop();
+		}
+		requestTimer = null;
+	}
+	
+	// Utility returns true when we have a position measurement stored in posDegrees
+	function hasPosition() {
+		return posDegrees != null && posDegrees.size() == 2;
+	}
+	
+	// Use this variable to rate limit my google maps API calls...
+    var webRequests = 0;
+    
+    // Ask Google Nearby Search to find the nearest Dunkin Donuts
+    // Documentation for nearby search here: https://developers.google.com/places/web-service/search#PlaceSearchResponses
+    function requestNearestDunkin() {
+    	// Make sure we have the current GPS coords needed to calculate the nearest dunkin donuts
+		if(!hasPosition()) {
+			// We don't know where we are, so we'll have to wait to find out
+			System.println("Where the heck are we?");
+			return;
+		}
+		
+		// START TEMPORARY SOLUTION //
+		if (webRequests < 10) {
+		System.println("Requesting nearby");
+		//////////////////////////////
+		
+		Comm.makeWebRequest( 
+			"https://maps.googleapis.com/maps/api/place/nearbysearch/json", 
+			{"location" => posDegrees[0] + "," + posDegrees[1],
+				"radius" => "10000",
+				"type" => "restaurant",
+				"name" => "dunkin+donuts",
+				"key" => API_KEY},
+			{:method => Comm.HTTP_REQUEST_METHOD_GET, 
+				:responseType => Comm.HTTP_RESPONSE_CONTENT_TYPE_JSON}, 
+			method(:nearbySearchCallback)
+		);
+		
+		// END TEMP SOLUTION //
+		webRequests++;
+		}
+		///////////////////////
+	}
+    
+    // Check the response and parse the returned data
+	function nearbySearchCallback(responseCode, data) {
+		if(isBadResponse(responseCode, data) ) {
+			nearestPlaceId = null;
+			
+			isOpen = false;
+			text = "None nearby";
+			donuts = 0;
+
+			Ui.requestUpdate();
+			
+			return false;
+		}
+		
+		// An Array of the nearby Dunkin Donuts restaurants. 
+		// We know that there are some: If not, then the status returned would be "ZERO_RESULTS"
+		var results = data["results"];
+		
+		var closestDunkin = null;
+		for(var i = 0; i < results.size(); i++) {
+			if((results[i]["name"]).find("Dunkin' Donuts") != null) {
+				//I think this is actually a dunkin, because the name contains Dunkin' Donuts
+				closestDunkin = results[i];
+				break;
+			} 
+			// Otherwise I don't think this is actually a dunkin. Find another
+		}
+
+		if(closestDunkin != null) {
+			nearestPlaceId = closestDunkin["place_id"];
+		
+			isOpen = closestDunkin["opening_hours"]["open_now"];
+			text = "Calculating...";
```

Comment:
I don't like this here. It flashes on screen for a quick moment every 15 seconds. remove it.